src/js/map/choropleth/subindicator_calculator.js

```diff
@@ -1,20 +1,27 @@
 import {percFmt, numFmt} from '../../utils'
 
 export default class SubindicatorCalculator {
-    calculate(args) {
-        if (typeof args.data === 'undefined') {
-            return null;
-        }
-
-        const result = Object.entries(args.data).map(([code, codeValue]) => {
-            const sumSubindicatorsValueForCode = args.subindicatorArr.reduce((sum, subindicator) => {
-                let childrenValue = subindicator.children[code] || 0;
-                return sum + childrenValue;
+    calculate(childData, primaryGroup, selectedSubindicator) {
+
+        const key = primaryGroup;
+        const val = selectedSubindicator;
+
+        const result = Object.entries(childData).map(([code, data]) => {
+            let filteredArr = data.filter((a) => {
+                return a[key] === val;
```

Comment:
can we rather use `primaryGroup` and `selectedSubindicator` here directly. it is more descriptive to what's going on.